genrl/deep/agents/dqn/base.py

```diff
@@ -0,0 +1,246 @@
+            self.batch_size
+        )
+        states = states.reshape(-1, *self.env.obs_shape)
+        actions = actions.reshape(-1, *self.env.action_shape).long()
+        rewards = torch.FloatTensor(rewards).reshape(-1)
+        next_states = next_states.reshape(-1, *self.env.obs_shape)
+        dones = torch.FloatTensor(dones).reshape(-1)
+
+        q_values = self.get_q_values(states, actions)
+        target_q_values = self.get_target_q_values(next_states, rewards, dones)
+        loss = F.mse_loss(q_values, target_q_values)
+        self.logs["value_loss"].append(loss.item())
+        return loss
+
+    def update_params(self, update_interval: int) -> None:
+        """Update parameters of the model
+
+        Args:
+            update_interval (int): Interval between successive updates of the target model
+        """
+        for timestep in range(update_interval):
+            loss = self.get_q_loss()
+            self.optimizer.zero_grad()
+            loss.backward()
+            self.optimizer.step()
+
+            if timestep % update_interval == 0:
+                self.update_target_model()
+
+    def calculate_epsilon_by_frame(self) -> float:
+        """Helper function to calculate epsilon after every timestep
+        """
+        return self.min_epsilon + (self.max_epsilon - self.min_epsilon) * np.exp(
+            -1.0 * self.timestep / self.epsilon_decay
+        )
+
+    def get_hyperparams(self) -> Dict[str, Any]:
+        """Get relevant hyperparameters to save
+
+        Returns:
+            hyperparams (:obj:`dict`): Hyperparameters to be saved
+        """
+        hyperparams = {
+            "gamma": self.gamma,
+            "batch_size": self.batch_size,
+            "lr": self.lr_value,
+            "replay_size": self.replay_size,
+            "weights": self.model.state_dict(),
+            "timestep": self.timestep,
+        }
+        return hyperparams
+
+    def load_weights(self, weights) -> None:
+        """Load weights for the agent from pretrained model
+        """
+        self.model.load_state_dict(weights["weights"])
+
+    def get_logging_params(self) -> Dict[str, Any]:
+        """Gets relevant parameters for logging
+
+        Returns:
+            logs (:obj:`dict`): Logging parameters for monitoring training
+        """
+        logs = {
+            "value_loss": safe_mean(self.logs["value_loss"]),
+            "epsilon": safe_mean(self.logs["epsilon"]),
+        }
+        self.empty_logs()
+        return logs
+
+    def empty_logs(self) -> None:
+        """Empties logs
+        """
+        self.logs = {}
+        self.logs["value_loss"] = []
+        self.logs["epsilon"] = []
+
+
+class DQN(BaseDQN):
+    """Vanilla DQN class
+
+    Simply calls the `create_model` function from the base class
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(DQN, self).__init__(*args, **kwargs)
+        self.empty_logs()
+        self.create_model()
```

Comment:
Same reason why I was suggesting to make the `BaseDQN` just `DQN` as this class does definition does not seem very useful.